File: 7a.py
#! /usr/bin/python3
import sys

f = open(sys.argv[1], "r")
crabs = [int(c) for c in f.readline().split(",")]
crabs.sort()

middle = crabs[len(crabs) // 2]
fuel = sum(map(lambda c: abs(middle - c), crabs))
print(fuel)

File: 7a.pl
#! /usr/bin/perl
use warnings;
use strict;
use feature qw{ say };

use ARGV::OrDATA;
use List::Util qw{ sum };

my @crabs = split /,/, <>;
chomp $crabs[-1];

@crabs = sort { $a <=> $b } @crabs;
my $middle = $crabs[ @crabs / 2 ];

my $fuel = sum(map abs($middle - $_), @crabs);
say $fuel;


__DATA__
16,1,2,0,4,2,7,1,2,14
